### PR TITLE
Add token-dependant region-specific api endpoint

### PR DESCRIPTION
**Pull request by tpinto**
Added region to configuration, which is derived from the api_key and is then used to setup the api endpoint

### CODE REVIEW

File: spec/onfido_spec.rb [hunk 1]
```diff
@@ -35,6 +35,27 @@
       end
     end
 
+    describe 'using a US API token' do
```
Comment (timrogers) [inline spec/onfido_spec.rb]: Can we add a spec for what happens if it's not an API key with a "region bit"?

File: lib/onfido/configuration.rb [hunk 2]
```diff
@@ -30,8 +30,26 @@ def logger
       RestClient.log ||= NullLogger.new
     end
 
+    def region
+      first_bit = api_key.split("_")[0]
+
+      case first_bit
```
Comment (timrogers) [inline lib/onfido/configuration.rb]: We could rewrite this to be more Ruby-like as:

```ruby
return if %w(live test).include?(first_bit)
first_bit
```

File: lib/onfido/configuration.rb [hunk 3]
```diff
@@ -30,8 +30,26 @@ def logger
...
+      when "live", "test" # it's an old, regionless api-token
+        nil
+      else # it's a new token the first bit being the region
+        first_bit
+      end
+
+    rescue
```
Comment (timrogers) [inline lib/onfido/configuration.rb]: What kind of error are we trying to rescue here?

File: spec/onfido_spec.rb (hunk 1 above)
Comment (tpinto) [reply spec/onfido_spec.rb]: There's a `describe 'using an old API token'` below to test using the old-style API keys (no region bit), is that what you mean? As per the other comments, I'm finishing some related updates and will be pushing them in a bit.

Comment (timrogers) [reply spec/onfido_spec.rb]: Ah, I didn't spot that test.

**Comment by timrogers** [comment]
@tpinto Are there still changes you're planning to make to this?

**Comment by tpinto** [comment]
@timrogers sorry, didn't realise you asked and kind of fell off my radar. I'm not planning any more changes, is this good to be merged and included in a new version? Thanks.

**Comment by timrogers** [comment]
@tpinto I'll merge this now; thanks. Does it make sense to release it now? Is the functionality available in the API?

**Comment by tpinto** [comment]
@timrogers no rush to release *now* if you plan to release in the next week or so. the functionality is either there (api.us.onfido.com is active and responding), being added (no tokens starting with eu_ or us_ have been issued yet) or being completed as we speak (api.eu.onfido.com will point to api.onfido.com shortly)

**Comment by timrogers** [comment]
Released as v0.11.0 🎉

**Pull request merged**